Source/WebCore/rendering/CSSFilter.cpp

```diff
@@ -84,25 +84,30 @@ static RefPtr<FilterEffect> createBlurEffect(const BlurFilterOperation& blurOper
 
 static RefPtr<FilterEffect> createBrightnessEffect(const BasicComponentTransferFilterOperation& componentTransferOperation)
 {
-    ComponentTransferFunction transferFunction;
-    transferFunction.type = FECOMPONENTTRANSFER_TYPE_LINEAR;
-    transferFunction.slope = narrowPrecisionToFloat(componentTransferOperation.amount());
-    transferFunction.intercept = 0;
+    auto colorMatrix = brightnessColorMatrix(componentTransferOperation.amount());
 
-    ComponentTransferFunction nullFunction;
-    return FEComponentTransfer::create(transferFunction, transferFunction, transferFunction, nullFunction);
+    Vector<float> inputParameters {
+        colorMatrix.at(0, 0), colorMatrix.at(0, 1), colorMatrix.at(0, 2), 0, 0,
```

Comment:
I think we can probably make a nice helper func on ColorMatrix<> to make these Vectors. (is this row major or column major?).

Reply:
I agree this will be a nice change. This conversion already exists in two places `createGrayScaleEffect()` and `createSepiaEffect()'.  This patch is adding four new ones.